From dc7ced690222a23c98ad4b8640635786ae9b3132 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Peter=20Od=C3=A9us?= <peter.odeus@gmail.com>
Date: Fri, 15 Jan 2010 13:29:46 +0100
Subject: [PATCH] temporary fix awaiting checkin to this branch. See http://sourceforge.net/tracker/?func=detail&aid=2912627&group_id=20024&atid=120024

---
 .../isomorphism/UniversalIsomorphismTester.java    |   29 ++++++++++++++-----
 1 files changed, 21 insertions(+), 8 deletions(-)

diff --git a/src/main/org/openscience/cdk/isomorphism/UniversalIsomorphismTester.java b/src/main/org/openscience/cdk/isomorphism/UniversalIsomorphismTester.java
index 4a93a68..4746df0 100644
--- a/src/main/org/openscience/cdk/isomorphism/UniversalIsomorphismTester.java
+++ b/src/main/org/openscience/cdk/isomorphism/UniversalIsomorphismTester.java
@@ -439,14 +439,27 @@ public class UniversalIsomorphismTester {
       // handle single query atom case separately
       if (g2.getAtomCount() == 1) {
           List<List<RMap>> matches = new ArrayList<List<RMap>>();
-          IQueryAtom queryAtom = (IQueryAtom) g2.getAtom(0);
-          for (IAtom atom : g1.atoms()) {
-              if (queryAtom.matches(atom)) {
-                  List<RMap> lmap = new ArrayList<RMap>();
-                  lmap.add(new RMap(g1.getAtomNumber(atom), 0));
-                  matches.add(lmap);
-              }
-          }
+          
+          
+          IAtom singleAtom = g2.getAtom(0);
+		  if (singleAtom instanceof IQueryAtom) {
+			  for (IAtom atom : g1.atoms()) {
+	              if (((IQueryAtom)singleAtom).matches(atom)) {
+	                  List<RMap> lmap = new ArrayList<RMap>();
+	                  lmap.add(new RMap(g1.getAtomNumber(atom), 0));
+	                  matches.add(lmap);
+	              }
+	          }
+		  } else {
+			  String atomSymbol = singleAtom.getSymbol();
+			  for(IAtom atom : g1.atoms()){
+				  if(atom.getSymbol().equals(atomSymbol)){
+					  List<RMap> lmap = new ArrayList<RMap>();
+	                  lmap.add(new RMap(g1.getAtomNumber(atom), 0));
+	                  matches.add(lmap);
+				  }
+			  }
+		  }
           return matches;
       }
 
-- 
1.6.5.3

